### PR TITLE
[T3A4][W11-A1]Tan Wang Leng

### PR DESCRIPTION
**Pull request by yamgent**

### CODE REVIEW

**Comment by yamgent** [comment]
## Notes
If the file somehow gets lost in the middle of the program operation, then the program is in an inconsistent state. Afaik, all types of inconsistencies in the program is followed with program crash (treated as unrecoverable), so file lost also crashes program due to FileNotFoundException followed by StorageOperationException.

**TLDR**: The crash is a feature not a bug :).

**Comment by yamgent** [comment]
Ready for review.

**Comment by yamgent** [comment]
The program will now attempt recovery by handling the FileNotFoundException and recreate the file (as per forum posting "About [LO-Exceptions]").

File: src/seedu/addressbook/storage/StorageFile.java [hunk 1]
```diff
@@ -84,6 +84,12 @@ private static boolean isValidPath(Path filePath) {
      * @throws StorageOperationException if there were errors converting and/or storing data to file.
      */
     public void save(AddressBook addressBook) throws StorageOperationException {
+        
+        try {
+            ensureStorageFileExists();
```
Comment (cmkumar87) [inline src/seedu/addressbook/storage/StorageFile.java]: consider a better name for this method. for e.g., doesFileExist()

File: src/seedu/addressbook/storage/StorageFile.java [hunk 2]
```diff
@@ -146,9 +147,16 @@ public AddressBook load() throws StorageOperationException {
...
-            save(empty);
-            return empty;
+            final AddressBook emptyAddressBook = new AddressBook();
+            
+            try {
+                createEmptyFile(path);
+            } catch (IOException ioe) {
+                throw new StorageOperationException("Error writing to file: " + path);
+            }
+
```
Comment (cmkumar87) [inline src/seedu/addressbook/storage/StorageFile.java]: indentations are way off. I remember commenting about coding style adherence in your PR last week too. This is easy to do with eclipse auto format. Please follow.

File: src/seedu/addressbook/storage/StorageFile.java [hunk 3]
```diff
@@ -146,9 +147,16 @@ public AddressBook load() throws StorageOperationException {
...
         // create empty file if not found
         } catch (FileNotFoundException fnfe) {
-            final AddressBook empty = new AddressBook();
-            save(empty);
-            return empty;
+            final AddressBook emptyAddressBook = new AddressBook();
+            
+            try {
+                createEmptyFile(path);
+            } catch (IOException ioe) {
```
Comment (cmkumar87) [inline src/seedu/addressbook/storage/StorageFile.java]: doesn't look right. refactor recovery to a helper method.

**Comment by cmkumar87** [comment]
@yamgent Please review comments and close the PR. We are now advised to follow this protocol.

**Comment by yamgent** [comment]
@cmkumar87 Um, can you please clarify regarding the indentation thing? The tabs seems pretty aligned to me (btw, it is a try statement *INSIDE* a try statement :P)

**Pull request closed without merging**